agent/api/task/task.go

```diff
@@ -846,6 +876,81 @@ func (task *Task) dockerHostConfig(container *apicontainer.Container, dockerCont
 	return hostConfig, nil
 }
 
+func (task *Task) awslogsEndpointOverride(hostConfig *docker.HostConfig, dockerVersion string) bool {
```

Comment:
Can you rename this to `shouldOverrideAWSLogsEndpoint` so that it conveys that it's just providing the logic for whether an override should be set?